include/rmm/exec_policy.hpp

```diff
@@ -38,4 +42,13 @@ inline auto exec_policy(cuda_stream_view stream             = cuda_stream_defaul
   return thrust::cuda::par(rmm::mr::thrust_allocator<char>(stream, mr)).on(stream.value());
 }
 
+struct exec_policy_t {
+  explicit exec_policy_t(cuda_stream_view stream) : stream_(stream) {}
+
+  operator thrust_exec_policy_t() { return exec_policy(stream_); }
```

Comment:
With the above change, you could return here `return thrust::cuda::par(rmm::mr::thrust_allocator<char>(stream, mr)).on(stream.value());`